### PR TITLE
Cherry pick more dedup

### PR DESCRIPTION
**Pull request by coolo**
I pulled back the evolution refactoring. This needs to be done more test driven - preferably by someone who knows the tests.

### CODE REVIEW

**Comment by coolo** [comment]
+174 −269

File: tests/x11/reboot_gnome.pm [hunk 1]
```diff
@@ -24,7 +24,7 @@ sub run() {
     if (get_var("SHUTDOWN_NEEDS_AUTH")) {
         assert_screen 'reboot-auth';
         wait_still_screen;
-        type_password;
+        type_string $testapi::password, max_interval => 5;
```
Comment (okurz) [inline tests/x11/reboot_gnome.pm]: should be save to rely on https://github.com/os-autoinst/os-autoinst/pull/636 now

File: tests/x11regressions/gnomecase/change_password.pm [hunk 2]
```diff
@@ -64,67 +64,48 @@ sub switch_user {
...
     wait_still_screen;
     send_key "alt-p";
     wait_still_screen;
-    type_string "$password";
+    type_password;
     wait_still_screen;
     send_key "alt-n";
     wait_still_screen;
-    type_string "$newpwd";
+    type_string $newpwd;
```
Comment (okurz) [inline tests/x11regressions/gnomecase/change_password.pm]: should be `type_password`, too, then

File: tests/x11/reboot_gnome.pm (hunk 1 above)
Comment (coolo) [inline tests/x11/reboot_gnome.pm]: you are really hardneckig!

**Pull request merged**